lib/puppet/type/resources.rb

```diff
@@ -145,14 +145,24 @@ def user_check(resource)
 
     return false if system_users.include?(resource[:name])
     return false if unless_uids && unless_uids.include?(current_uid)
-
-    current_uid > self[:unless_system_user]
+    if current_uid.is_a?(String)
+      # Windows user; is a system user if any regex matches.
+      not windows_system_sid_regexes.any? { |regex| current_uid =~ regex }
+    else
+      current_uid > self[:unless_system_user]
+    end
   end
 
   def system_users
     %w{root nobody bin noaccess daemon sys}
   end
 
+  def windows_system_sid_regexes
+    # Administrator, Guest, Domain Admins, Schema Admins, Enterprise Admins.
```

Comment:
Reference here to where these users were obtained

Reply:
Added.